### PR TITLE
tpm2: Split up SealedKeyObject

### PR DESCRIPTION
**Pull request by chrisccoulson**
SealedKeyObject will be deprecated by a follow up PR, and a new API
will be introduced to replace it. This splits it into code that can be
reused by the new API (sealedKeyDataBase) and moves the existing
API to a new file.

### CODE REVIEW

File: tpm2/policy_v3.go [hunk 1]
```diff
@@ -44,6 +44,10 @@ func computeV3PcrPolicyRefFromCounterContext(context tpm2.ResourceContext) tpm2.
 	return computeV1PcrPolicyRefFromCounterContext(context)
 }
 
+func computeV3PcrPolicyCounterAuthPolicies(alg tpm2.HashAlgorithmId, updateKeyName tpm2.Name) tpm2.DigestList {
```
Comment (pedronis) [inline tpm2/policy_v3.go:47]: this is not used yet?

File: tpm2/update_test.go [hunk 2]
```diff
@@ -1,153 +0,0 @@
...
-		PCRPolicyCounterHandle: s.NextAvailableHandle(c, 0x01810000)})
-	c.Check(err, ErrorMatches, "invalid key data: cannot complete authorization policy assertions: the PCR policy has been revoked")
-}
-
-func (s *updateSuite) TestRevokeOldPCRProtectionPoliciesWithoutPCRPolicyCounter(c *C) {
-	err := s.testRevokeOldPCRProtectionPolicies(c, &KeyCreationParams{
-		PCRProfile:             tpm2test.NewPCRProfileFromCurrentValues(tpm2.HashAlgorithmSHA256, []int{7, 23}),
-		PCRPolicyCounterHandle: tpm2.HandleNull})
-	c.Check(err, IsNil)
-}
```
Comment (pedronis) [inline tpm2/update_test.go:153]: what happened to these tests?

File: tpm2/keydata.go
Comment (pedronis) [inline tpm2/keydata.go:1]: A bit unrelated and it's true also on master but I notice this had and the split parts have relatively low coverage

**Review by pedronis — COMMENTED** [review]
did a pass, as it's mostly a reorg just some small comments

File: tpm2/policy_v3.go (hunk 1 above)
Comment (chrisccoulson) [reply tpm2/policy_v3.go]: I've moved this to the follow up PR.

File: tpm2/update_test.go (hunk 2 above)
Comment (chrisccoulson) [reply tpm2/update_test.go:153]: Ah, I missed those when I was moving things around. I've added them back again.

**Review by pedronis — APPROVED** [review]
thanks

**Pull request merged**